File: TypingAccuracy.js

function TypingAccuracy(sentence , typedWord, timeTaken){

let words = sentence.split();
let typedWords = typedWord.split();
let count = 0;
let correct = 0;
if(typedWords.length == words.length){

for(word of words){
	
	if(word == typedWords[count]){correct += 1;}
count += 1;
}

}

else{console.log("Incorrect sentence");}


let accuracy = correct / count * 100;


wordPerMinute = count / (timeTaken / 60);

return{
accurate : accuracy,
word : wordPerMinute
}
}


const sentence = "I love programming";
const prompt = require('prompt-sync')();

const start = Date.now();
let typed = prompt(`Enter the sentence '${sentence}' : `);
const end = Date.now();

const timeTaken = (start - end) / 1000;

TypingAccuracy(sentence , typed , timeTaken)

console.log(`Timetake: ${timeTaken * -1}`);
console.log(`Words per minute: ${TypingAccuracy(sentence , typed , timeTaken).word}`);
console.log(`Typing accuracy: ${TypingAccuracy(sentence , typed , timeTaken).accurate}`);

File: typing_accuracy.py
import time
def get_typing_accuracy(sentence , typed):
	
	time_taken_to_type = start_point - end_point
	
	words = sentence_to_type.split()

	typed_words = typed_input.split()
	
	correct_words = 0
	index = 0
	if len(words) == len(typed_words): 
		for word in words:
			if word == typed_words[index]:
				correct_words += 1
			index += 1
	else:
		print("The sentence is wrong!")
		
	
	accuracy = correct_words / len(words) * 100
	if time_taken_to_type > 0:
		word_per_minute = words / (time_taken_to_type / 60)
	else:
		word_per_minute = 0
	
	return time_taken_to_type, word_per_minute , accuracy	





sentence_to_type = "Anime is amazing isn't it?"
	
start_point = time.time()
typed_input = input(f"Enter the sentence '{sentence_to_type}' : ");
end_point = time.time()

typing = get_typing_accuracy(sentence_to_type , typed_input)

print(f"You took {typing[0] * -1:.2f} seconds to type~")
print(f"Words_per_minute {typing[1]:.2f}")
print(f"Typing accuracy = {typing[2]:.2f}%")